jaeger_client/utils.py

```diff
@@ -65,27 +64,56 @@ def local_ip():
     except IOError:
         ip = socket.gethostbyname('localhost')
     if ip.startswith('127.'):
-        # Check eth0, eth1, eth2, en0, ...
-        interfaces = [
-            i + bytes(n) for i in (b'eth', b'en', b'wlan') for n in range(3)
-        ]  # :(
-        for interface in interfaces:
-            try:
-                ip = interface_ip(interface)
+        ip = get_local_ip_by_interfaces()
+        if ip is None:
+            ip = get_local_ip_by_socket()
+            if ip is None:
+                ip = 'Unknown IP'
```

Comment:
maybe it's better to skip the IP tag in this case? Returning invalid string as IP might break something downstream.